addOns/graphql/src/main/java/org/zaproxy/addon/graphql/ImportFromFileDialog.java

```diff
@@ -87,7 +89,17 @@ protected boolean importDefinition() {
          * Import schema definition from file path; and
          * Send it to the parser.
          */
-        showWarningDialog(Constant.messages.getString("graphql.importfromdialog.message"));
-        return false;
+        File file = new File(getSchemaField().getText());
+
+        if (!file.canRead()) {
+            showWarningDialog(Constant.messages.getString(MESSAGE_PREFIX + "badfile"));
+            return false;
+        }
+        try {
+            GraphQlParser.parse(FileUtils.readFileToString(file, "UTF-8"), getEndpointUri());
```

Comment:
This will have to be done in a separate thread (guess `GraphQlParser` could be responsible for that), otherwise it will block the EDT.
Same when importing from URL.